mscore/inspector/inspector.cpp

```diff
@@ -160,13 +160,10 @@ void Inspector::setElements(const QList<Element*>& l)
 
             bool sameTypes = true;
             for (Element* ee : _el) {
-                  if (_element->type() != ee->type())
+                  if ((_element->type() != ee->type()) && // different and
+                      (!_element->isSlurTieSegment() && !ee->isSlurTieSegment()) || // neither Slur nor Tie either side, or
+                      (ee->isNote() && toNote(ee)->chord()->isGrace() != toNote(_element)->chord()->isGrace())) // HACK
```

Comment:
That line was introduced here: https://github.com/musescore/MuseScore/commit/df675154e5
for issue: https://musescore.org/en/node/33341
I think it is used to differentiate between normal and grace notes because of different properties such as leading/trailing space as seen in the issue.
On second thought, I think the leading space (not editable for grace notes) is the only difference between them.

With this proposed change, if (_element->type() != ee->type()) is true, then the following is not evaluated at all, right? Otherwise the two toNote(ee) and toNote(_element) could cause problems if the two types are different.